core-java-modules/core-java-collections-6/src/test/java/com/baeldung/nullcheck/NullValueCheckUnitTest.java

```diff
@@ -0,0 +1,92 @@
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.Test;
+
+class NullValueCheckUnitTest {
+
+    Function<Collection<?>, Long> countNulls = collection -> collection.stream()
+        .filter(Objects::isNull)
+        .count();
+
+    @Test
+    void givenList_whenNullValueAdded_doesNotFail() {
+        // adding nulls to a list
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        List<Integer> numbers = Arrays.asList(numberArray);
+        assertEquals(countNulls.apply(numbers), 3);
+
+        // accessing nulls from a list
+        assertNull(numbers.get(0));
+
+        // dereferencing nulls from a list
+        assertThrows(NullPointerException.class, () -> numbers.get(0)
+            .toString());
+    }
+
+    @Test
+    void givenHashSet_whenNullValueAdded_doesNotFail() {
+        // adding nulls to a HashSet
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        Set<Integer> numbers = new HashSet<>(Arrays.asList(numberArray));
+        assertEquals(countNulls.apply(numbers), 1);
+
+        // accessing null from a set
+        assertTrue(numbers.contains(null));
+
+        // dereferencing nulls from a set
+        assertThrows(NullPointerException.class, () -> numbers.forEach(item -> item.toString()));
+    }
+
+    @Test
+    void givenTreeSet_whenNullValueAdded_mightFail() {
+        // adding nulls to a TreeSet
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        assertThrows(NullPointerException.class, () -> new TreeSet<>(Arrays.asList(numberArray)));
+    }
+
+    @Test
+    void givenHashMap_whenNullKeyValueAdded_doesNotFail() {
+        // adding nulls to key value pairs
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        Map<Integer, Integer> numbers = new HashMap<>();
+        Arrays.stream(numberArray)
+            .forEach(integer -> numbers.put(integer, integer));
+        assertEquals(countNulls.apply(numbers.keySet()), 1);
+        assertEquals(countNulls.apply(numbers.values()), 1);
+
+        // accessing nulls from a map
+        assertTrue(numbers.containsKey(null));
+        assertTrue(numbers.containsValue(null));
+        assertNull(numbers.get(null));
+
+        // dereferencing nulls from a map
+        assertThrows(NullPointerException.class, () -> numbers.get(null)
+            .toString());
+    }
+
+    @Test
+    void givenTreeMap_whenNullKeyValueAdded_fails() {
+        // adding nulls to key value pairs
+        Integer[] numberArray = { null, 0, 1, null, 2, 3, null };
+        Map<Integer, Integer> numbers = new TreeMap<>();
+        assertThrows(NullPointerException.class, () -> Arrays.stream(numberArray)
```

Comment:
Refactor the code of the lambda to have only one invocation possibly throwing a runtime exception.

Reply:
We are verifying that a `NullPointerException` is thrown when attempting to insert a `null` key into a TreeMap. I have updated the test. Please review.